### PR TITLE
block_hotplug: replace device.hotplug() to vm.hotplug() to adapt q35

### PR DESCRIPTION
**Pull request by vivianQizhu**
Depends on https://github.com/avocado-framework/avocado-vt/pull/1558
id: 1532521
Signed-off-by: qizhu <qizhu@redhat.com>

### CODE REVIEW

**Comment by jingzhao84** [comment]
Ack

PASS 3-Host_RHEL.m7.u6.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.block_hotplug.block_virtio.fmt_qcow2.with_plug.with_reboot.multi_pci.q35
PASS 1-Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.block_hotplug.block_virtio.fmt_qcow2.with_plug.with_reboot.multi_pci.q35

File: qemu/tests/block_hotplug.py [hunk 1]
```diff
@@ -126,7 +124,7 @@ def unplug_device(vm, get_disk_cmd, device):
                 drive.set_param("file", find_image(img_list[num + 1]))
```
Comment (ldoktor) [inline qemu/tests/block_hotplug.py]: Not related to this patch but there is a reason the `virttest.qemu_devices.QContainer.images_define_by_variables` was developed and that is to automatically create controllers and detect proper device. The `QRHDrive` is only supported on RHEL but other systems can still use the `QHPDrive`...

Comment (vivianQizhu) [reply qemu/tests/block_hotplug.py]: @ldoktor Yes, I will contact the owner of this feature in our group to see if this can be refactored. Thanks.

Comment (ldoktor) [reply qemu/tests/block_hotplug.py]: @vivianQizhu actually a guy from Oracle already noticed that and tried to add QHPDrive support. I replied with `images_define_by_variables` proof-of-concept here: https://github.com/autotest/tp-qemu/pull/1384#pullrequestreview-121494572 it just needs some polishing and a bit of testing, but in general works.

Comment (vivianQizhu) [reply qemu/tests/block_hotplug.py]: @ldoktor Great, we will follow up it, thanks.

**Comment by vivianQizhu** [comment]
Close this one, it is all covered by #1396

**Pull request closed without merging**